hyperdome_client/add_server_dialog.py

```diff
@@ -100,45 +84,60 @@ def __init__(self, common, add_server_action):
         self.server_dialog_layout = QtWidgets.QVBoxLayout()
         self.server_dialog_layout.addWidget(self.server_add_text)
         self.server_dialog_layout.addWidget(self.server_nick_text)
-        self.server_dialog_layout.addLayout(self.radio_buttons)
+        self.server_dialog_layout.addLayout(radio_buttons)
         self.server_dialog_layout.addLayout(self.counselor_credentials)
         self.server_dialog_layout.addWidget(self.add_server_button)
 
         self.setLayout(self.server_dialog_layout)
 
-    def _make_server_from_fields(self):
-        """
-        Take text fields and package into server object to pass.
-        """
-        url = self.server_add_text.text()
-        nick = self.server_nick_text.text()
-        uname = self.counselor_username_input.text()
-        passwd = self.counselor_password_input.text()
-        is_counselor = self.is_counselor
-
-        return Server(
-            url=url, nick=nick, uname=uname, passwd=passwd, is_counselor=is_counselor
-        )
-
-    def radio_switch(self, radio_switch):
+    def radio_switch(self, is_toggled):
         """
         Show or hide crediential fields based on user type selected.
         """
-        if radio_switch.text() == "Counselor":
-            self.is_counselor = True
-            self.counselor_username_input.show()
-            self.counselor_password_input.show()
-        else:
-            self.is_counselor = False
-            self.counselor_username_input.hide()
-            self.counselor_password_input.hide()
+        self.is_counselor = is_toggled
+        self.counselor_credentials.setEnabled(is_toggled)
+        self.counselor_username_input.setVisible(is_toggled)
+        self.counselor_password_input.setVisible(is_toggled)
 
-    def closeEvent(self, event):
+    def add_server(self):
         """
-        Cleanup for when window is closed.
+        Reciever for the add server dialog to handle the new server details.
         """
-        self.counselor_username_input.clear()
-        self.counselor_password_input.clear()
-        self.server_add_text.clear()
+        try:
+            self.server = Server(
+                url=self.server_add_text.text(),
+                nick=self.server_nick_text.text(),
+                uname=self.counselor_username_input.text(),
+                passwd=self.counselor_password_input.text(),
+                is_counselor=self.is_counselor,
+            )
+        except Server.InvalidOnionAddress:
+            self.error_message.setText("Invalid onion address!")
+            self.error_message.exec_()
+            return
+
+        self.add_server_button.setEnabled(False)
+        self.add_server_button.setText("Checking...")
+
+        probe = threads.ProbeServerTask(self.session, self.server)
+        probe.signals.success.connect(self.set_server)
+        probe.signals.error.connect(self.bad_server)
+        self.worker.start(probe)
+
+    @QtCore.pyqtSlot(str)
+    def set_server(self, _: str):
```

Comment:
:eyes: 
`_: str` makes me incredibly uncomfortable

Reply:
Maybe it's totally fine, it's just
Wow

Reply:
well I wanted to communicate that the function prototype takes a string, but we don't care what it is. I guess the type hint is really unnecessary but I like it. To be clear the pyqtSlot needs to indicate that it's a string because it's creating a C++ function binding.